dali/imgcodec/decoders/libtiff/tiff_libtiff.cc

```diff
@@ -212,17 +223,23 @@ DecodeResult LibTiffDecoderInstance::Decode(SampleView<CPUBackend> out, ImageSou
   TensorShape<> out_line_shape = {roi.shape()[1], out_channels};
   TensorShape<> out_line_strides = kernels::GetStrides(out_line_shape);
 
-  for (int64_t roi_y = 0; roi_y < roi.shape()[0]; roi_y++) {
-    LIBTIFF_CALL(TIFFReadScanline(tiff.get(), row_in, roi.begin[0] + roi_y, 0));
+  TYPE_SWITCH(out.type(), type2id, OutType, (IMGCODEC_TYPES), (
+    VALUE_SWITCH(info.bit_depth, Depth, (8, 16, 32), (
+      using InputType = detail::depth2type<Depth>::type;
+      auto *row_in  = static_cast<InputType *>(row_buf.get());
+      OutType *const img_out = out.mutable_data<OutType>();
+      for (int64_t roi_y = 0; roi_y < roi.shape()[0]; roi_y++) {
```

Comment:
I found the name `roi_y` confusing in this context. In my head `roi_y` would go from roi.begin to roi.end. And in this case it goes from 0 to (roi.end-roi.begin). I would rather call this simply `y` to avoid confusion.

Reply:
I actually changed the name from `y` to `roi_y`, becase `y` was confusing for me :laughing: . For me `y` suggested that I'm iterating through the whole image (that is for example `y=0` means I'm processing the first row of the image etc). And `roi_y` (for me) indicates that I'm iterating only over the ROI, and the indexing is ROI-relative.

Reply:
I'll change it back to `y` in a moment